src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -135,6 +135,23 @@ public function create_item_permissions_check( $request ) {
 			);
 		}
 
+		// If the upload is an image, check if the server can handle the mime type.
+		$files = $request->get_file_params();
+		if (
+			! empty( $files ) &&
+			isset( $files['file'] ) &&
+			str_starts_with( $files['file']['type'], 'image/' )
+		) {
+			// Check if the image editor supports the type.
+			if ( ! wp_image_editor_supports( array( 'mime_type' => $files['file']['type'] ) ) ) {
```

Comment:
Could a test be added for this?

Reply:
Yes, probably. I will give it a try.

Reply:
added test in https://github.com/WordPress/wordpress-develop/pull/8322/commits/eb41d06d00f62def6520d5a12b8b1aef10cdca25 - I am currently unable to test this locally as my travel setup supports AVIF, hopefully tests work in CI.